test/benchmark_http_client_test.cc

```diff
@@ -310,8 +349,48 @@ TEST_F(BenchmarkClientHttpTest, BadContentLength) {
   };
 
   EXPECT_CALL(stream_encoder_, encodeData(_, _)).Times(0);
-  testBasicFunctionality(1, 1, 1);
+
+  TestBenchmarkClientProcessesExpectedInflightRequests(1, 1, 1, request_generator);
   EXPECT_EQ(1, getCounter("http_2xx"));
 }
 
+TEST_F(BenchmarkClientHttpTest, RequestGeneratorProvidingDifferentPathsSendsRequestsOnThosePaths) {
+  std::vector<HeaderMapPtr> requests_for_generator_to_send;
+  const std::initializer_list<std::pair<std::string, std::string>> header_map_for_first_request{
+      {":scheme", "http"},
+      {":method", "GET"},
+      {":path", "/a"},
+      {":host", "localhost"},
+      {"Content-Length", "1313"}};
+  const std::initializer_list<std::pair<std::string, std::string>> header_map_for_second_request{
+      {":scheme", "http"},
+      {":method", "GET"},
+      {":path", "/b"},
+      {":host", "localhost"},
+      {"Content-Length", "1313"}};
+  requests_for_generator_to_send.push_back(
+      std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(header_map_for_first_request));
+  requests_for_generator_to_send.push_back(
+      std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(header_map_for_second_request));
+  std::vector<HeaderMapPtr>::iterator request_iterator;
+  request_iterator = requests_for_generator_to_send.begin();
+  RequestGenerator request_generator = [&]() {
+    auto item = *request_iterator;
+    request_iterator++;
+    return std::make_unique<RequestImpl>(item);
```

Comment:
collapse this to just `return std::make_unique<RequestImpl>(*request_iterator++);` ?

Reply:
Thanks, that was sloppy of me. Done.